sleap/nn/inference.py

```diff
@@ -5300,71 +5301,104 @@ def _make_provider_from_cli(args: argparse.Namespace) -> Tuple[Provider, str]:
 
     data_path_obj = Path(data_path)
 
+    # Set output_path_list to None as a default to return later
+    output_path_list = None
+
     # Check that input value is valid
     if not data_path_obj.exists():
         raise ValueError("Path to data_path does not exist")
 
-    # Check for multiple video inputs
-    # Compile file(s) into a list for later itteration
-    if data_path_obj.is_dir():
-        data_path_list = []
-        for file_path in data_path_obj.iterdir():
-            if file_path.is_file():
-                data_path_list.append(Path(file_path))
     elif data_path_obj.is_file():
-        data_path_list = [data_path_obj]
+        # If the file is a CSV file, check for data_paths and output_paths
+        if data_path_obj.suffix.lower() == ".csv":
+            try:
+                # Read the CSV file
+                df = pandas.read_csv(data_path)
+
+                # collect data_paths from column
+                if "data_path" in df.columns:
+                    raw_data_path_list = df["data_path"].tolist()
+                else:
+                    raise ValueError(
+                        f"Column 'data_path' does not exist in the CSV file: {data_path}"
+                    )
```

Comment:
Design choice: we may want to be less sensitive to column naming. It's probably fair to assume that the first column of the CSV is the input path and the second column is the output path.

It'd be great to add some logic to auto-detect if the column names are present in the first row or not, and ignore it appropriately. An easy way would be to just do a `Path(df.iloc[0, 0]).exists()` and if not (or if it's not a path -- not sure if pathlib does any checking), then assume that it's a column name row and skip it.